File: Mathematics/sieve_of_eratosthenes.cpp
// CPP Code of Sieve Of Eratosthenes

#include <bits/stdc++.h>
using namespace std;

int *findPrime(int num) {

    int *prime = new int[num+1];

    prime[0] = 0, prime[1] = 0, prime[2] = 1;

    for(int i = 3; i <= num; ++i) {

        if(i%2) {

            prime[i] = 1;
        }
        else {

            prime[i] = 0;
        }
    }

    for(int i = 2; i <= num; ++i) {
        
        if(prime[i]) {
            
            for(int j = i*i; j <= num ; j += i) {

                prime[j] = 0;
            }
        }
    }

    return prime;
}

void showPrime(int *primeNum, int num) {

    for(int i = 0; i <= num; ++i) {
        
        if(primeNum[i]) {
            
            cout<<i<<" ";
        }
    }
}

int main() {

    int num;

    cin>>num;

    int *primeNum;

    primeNum = findPrime(num);

    showPrime(primeNum, num);

    return 0;
}

File: Mathematics/sieve_of_eratosthenes.py
#Python Code of Sieve Of Eratosthenes

def findPrime(num) :

    primeNum = [True for index in range(num+1)]
    
    index = 2

    while index <= num :

        if primeNum[index] :

            for index1 in range(index*index, num+1, index) :
                primeNum[index1] = False
        index = index +1
    
    
    return primeNum

def showPrime(prime, num) :

    for index in range(2, num+1):
        
        if prime[index] :
        
            print(index)

def main():

    num = int(input())

    prime = findPrime(num)

    showPrime(prime, num)

if __name__ == "__main__" :
    main()